core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/aggregate/StatisticalAggregateFunction.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eclipse RDF4J contributors.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.algebra.evaluation.function.aggregate;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.datatypes.XMLDatatypeUtil;
+import org.eclipse.rdf4j.query.BindingSet;
+import org.eclipse.rdf4j.query.QueryEvaluationException;
+import org.eclipse.rdf4j.query.algebra.evaluation.ValueExprEvaluationException;
+import org.eclipse.rdf4j.query.parser.sparql.aggregate.AggregateFunction;
+
+/**
+ * {@link AggregateFunction} used for processing of extended statistical aggregate operations through SPARQL.
+ *
+ * @author Tomas Kovachev t.kovachev1996@gmail.com
+ */
+public class StatisticalAggregateFunction extends AggregateFunction<StatisticCollector, Value> {
```

Comment:
Could you mark this as experimental to allow us to change things i the future if we need?